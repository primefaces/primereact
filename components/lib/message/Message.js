import React, { forwardRef, memo } from 'react';
import PropTypes from 'prop-types';
import { ObjectUtils, classNames } from '../utils/Utils';

export const Message = memo(forwardRef((props, ref) => {

    const createContent = () => {
        if (props.content) {
            return ObjectUtils.getJSXElement(props.content, props);
        }

        const text = ObjectUtils.getJSXElement(props.text, props);
        const icon = classNames('p-inline-message-icon pi', {
            'pi-info-circle': props.severity === 'info',
            'pi-exclamation-triangle': props.severity === 'warn',
            'pi-times-circle': props.severity === 'error',
            'pi-check': props.severity === 'success',
        });

        return (
            <>
                <span className={icon}></span>
                <span className="p-inline-message-text">{text}</span>
            </>
        )
    }

    const className = classNames('p-inline-message p-component', {
        'p-inline-message-info': props.severity === 'info',
        'p-inline-message-warn': props.severity === 'warn',
        'p-inline-message-error': props.severity === 'error',
        'p-inline-message-success': props.severity === 'success',
        'p-inline-message-icon-only': !props.text
    }, props.className);
    const content = createContent();

    return (
        <div id={props.id} aria-live="polite" className={className} style={props.style} role="alert">
            {content}
        </div>
    )
}));

Message.defaultProps = {
    __TYPE: 'Message',
    id: null,
    className: null,
    style: null,
    text: null,
    severity: 'info',
    content: null
}

Message.propTypes /* remove-proptypes */ = {
    __TYPE: PropTypes.string,
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    text: PropTypes.any,
    severity: PropTypes.string,
    content: PropTypes.any
};
