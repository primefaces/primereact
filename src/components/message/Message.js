import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ObjectUtils, classNames } from '../utils/Utils';

export class Message extends Component {

    static defaultProps = {
        id: null,
        className: null,
        style: null,
        text: null,
        severity: 'info',
        content: null
    }

    static propTypes = {
        id: PropTypes.string,
        className: PropTypes.string,
        style: PropTypes.object,
        text: PropTypes.string,
        severity: PropTypes.string,
        content: PropTypes.any
    };

    getContent() {
        if (this.props.content) {
            return ObjectUtils.getJSXElement(this.props.content, this.props);
        }

        const icon = classNames('p-inline-message-icon pi', {
            'pi-info-circle': this.props.severity === 'info',
            'pi-exclamation-triangle': this.props.severity === 'warn',
            'pi-times-circle': this.props.severity === 'error',
            'pi-check': this.props.severity === 'success',
        });

        return (
            <>
                <span className={icon}></span>
                <span className="p-inline-message-text">{this.props.text}</span>
            </>
        );
    }

    render() {
        const className = classNames('p-inline-message p-component', {
            'p-inline-message-info': this.props.severity === 'info',
            'p-inline-message-warn': this.props.severity === 'warn',
            'p-inline-message-error': this.props.severity === 'error',
            'p-inline-message-success': this.props.severity === 'success',
            'p-inline-message-icon-only': !this.props.text
        }, this.props.className);

        const content = this.getContent();

        return (
            <div id={this.props.id} aria-live="polite" className={className} style={this.props.style} role="alert">
                {content}
            </div>
        );
    }
}
