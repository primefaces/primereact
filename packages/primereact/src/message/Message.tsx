'use client';
import { Component } from '@primereact/core/component';
import { useMessage } from '@primereact/headless/message';
import { styles } from '@primereact/styles/message';
import { mergeProps } from '@primeuix/utils';
import { withComponent } from 'primereact/base';
import * as React from 'react';
import { MessageProvider } from './Message.context';
import { defaultProps } from './Message.props';
import { MessageClose } from './close/MessageClose';
import { MessageContent } from './content/MessageContent';
import { MessageIcon } from './icon/MessageIcon';
import { MessageText } from './text/MessageText';

export const Message = withComponent({
    name: 'Message',
    defaultProps,
    styles,
    setup(instance) {
        const message = useMessage(instance.inProps);

        return message;
    },
    render(instance) {
        const { id, props, ptmi, cx, state } = instance;

        const rootProps = mergeProps(
            {
                id,
                className: cx('root'),
                ...(!state.visible && {
                    style: {
                        display: 'none'
                    }
                })
            },
            ptmi('root')
        );

        return (
            <MessageProvider value={instance}>
                <Component instance={instance} attrs={rootProps} children={props.children} />
            </MessageProvider>
        );
    },
    components: {
        Content: MessageContent,
        Text: MessageText,
        Icon: MessageIcon,
        Close: MessageClose
    }
});
