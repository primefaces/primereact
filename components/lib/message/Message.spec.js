import '@testing-library/jest-dom';
import { Message } from './Message';

import { snapshot } from '../../test';

describe('Message', () => {
    snapshot(<Message />, 'default');
    snapshot(<Message severity="success" text="Jest" />, 'severity success');
    snapshot(<Message severity="info" text="Jest" />, 'severity info');
    snapshot(<Message severity="warn" text="Jest" />, 'severity warn');
    snapshot(<Message severity="error" text="Jest" />, 'severity error');
    snapshot(<Message icon="pi pi-check" />, 'icon');
    snapshot(<Message content={<span>Test</span>} />, 'content');
    snapshot(
        <Message>
            <span>Jester</span>
        </Message>,
        'templating'
    );
});
