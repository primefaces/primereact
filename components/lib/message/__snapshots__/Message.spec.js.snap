// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message content 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-info p-inline-message-icon-only"
    role="alert"
  >
    <span>
      Test
    </span>
  </div>
</div>
`;

exports[`Message default 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-info p-inline-message-icon-only"
    role="alert"
  >
    <span
      class="p-inline-message-icon pi pi-info-circle"
    />
    <span
      class="p-inline-message-text"
    />
  </div>
</div>
`;

exports[`Message icon 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-info p-inline-message-icon-only"
    role="alert"
  >
    <span
      class="p-inline-message-icon pi pi-check"
    />
    <span
      class="p-inline-message-text"
    />
  </div>
</div>
`;

exports[`Message severity error 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-error"
    role="alert"
  >
    <span
      class="p-inline-message-icon pi pi-times-circle"
    />
    <span
      class="p-inline-message-text"
    >
      Jest
    </span>
  </div>
</div>
`;

exports[`Message severity info 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-info"
    role="alert"
  >
    <span
      class="p-inline-message-icon pi pi-info-circle"
    />
    <span
      class="p-inline-message-text"
    >
      Jest
    </span>
  </div>
</div>
`;

exports[`Message severity success 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-success"
    role="alert"
  >
    <span
      class="p-inline-message-icon pi pi-check"
    />
    <span
      class="p-inline-message-text"
    >
      Jest
    </span>
  </div>
</div>
`;

exports[`Message severity warn 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-warn"
    role="alert"
  >
    <span
      class="p-inline-message-icon pi pi-exclamation-triangle"
    />
    <span
      class="p-inline-message-text"
    >
      Jest
    </span>
  </div>
</div>
`;

exports[`Message templating 1`] = `
<div>
  <div
    aria-live="polite"
    class="p-inline-message p-component p-inline-message-info p-inline-message-icon-only"
    role="alert"
  >
    <span
      class="p-inline-message-icon pi pi-info-circle"
    />
    <span
      class="p-inline-message-text"
    />
  </div>
</div>
`;
